File: src/com/mcblox/parkour/cmds/parkour/CmdHelp.java
package com.mcblox.parkour.cmds.parkour;

import static org.bukkit.ChatColor.*;

import java.util.List;

import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import com.mcblox.parkour.enums.ParkourSubCommands;
import com.mcblox.parkour.objects.BloxCommand;

public class CmdHelp extends BloxCommand {

	@Override
	public void execute(CommandSender sender, String[] args) {
		
		for (ParkourSubCommands psc : ParkourSubCommands.values()) {
			sender.sendMessage(RED + psc.getCmd().getUsage() + GRAY + " - " + getDescription());
		}
		
		//-- End: execute(CommandSender, String[])
	}

	@Override
	public List<String> tabComplete(Player player, String[] args) {
		return null;
	}

	@Override
	public String getLabel() {
		return "HELP";
	}


	@Override
	public String getPermission() {
		return "parkour.help";
	}

	@Override
	public String getUsage() {
		return "/parkour help";
	}

	@Override
	public String getDescription() {
		return "Shows the list of parkour commands";
	}
}

File: src/org/serversmc/parkour/commands/CmdHelp.kt
package org.serversmc.parkour.commands

import org.bukkit.command.*
import org.bukkit.entity.*
import org.serversmc.interfaces.*

object CmdHelp : ICommand {
	
	override fun execute(sender: CommandSender, args: MutableList<out String>) {
	}
	
	override fun tabComplete(player: Player, args: MutableList<out String>): MutableList<String>? = ArrayList()
	override fun getDescription(): String = ""
	override fun getLabel(): String = ""
	override fun getPermission(): String = ""
	override fun getUsage(): String = ""
	
}